### ISSUE TITLE
Is it possible to make the normal and regular parse command formats consistent

### ISSUE THREAD

**Issue by Silentdoer** [nushell/nushell#7459]
### Describe the solution you'd like

normal parse:
```nu
echo "hi there" | parse "{foo} {bar}"
```
regex parse:
```nu
echo "hi there" | parse -r '(?<foo>\w+) (?<bar>\w+)'
```

Is it possible to change the normal parse command format to `parse "<foo> <bar>`?
and simplify the parse command format for regular expressions `parse -r '(<foo>\w+)\s+(<bar>\w+)'` or `parse -r '<foo#\w+>\s+<bar#\w+>'`

**Comment by fdncred** [comment]
We could make `{foo} {bar}` be `<foo> <bar>`. It's debatable whether that's more helpful

We probably wouldn't change how regular expressions work because wouldn't we have to reinvent a new regular expression dialect?

The `parse` command uses named capture groups as column names. `(?<foo>blah)` is a standard way to create named capture groups. The other dialect that we support is `(?P<foo>blah)`. I don't believe either of your suggestions are valid regular expressions `parse -r '(<foo>\w+)\s+(<bar>\w+)'` or `'<foo#\w+>\s+<bar#\w+>'`.

**Comment by Silentdoer** [comment]
Yes, the regular expression grouping is something I didn't think of, but should the normal parse be changed to `<column>`? Because split column are also grouped with `<>`

**Comment by fdncred** [comment]
> Yes, the regular expression grouping is something I didn't think of

This is the only way that parse works at this point. You have to use a regular expression capture group in order to get results in a table with column names. Like I said, if we changed it, we would somehow have to define some new non-regex syntax, parse what you want the column name to be, then we'd have to change that into a capture group, then use it. I'm not a fan of that right now.

**Comment by 132ikl** [comment]
since this is more of a proposal i'm going to convert this to a discussion, in service of trying to minimize "forever" issues in the issue tracker
